framework/include/outputs/ProgressOutput.h

```diff
@@ -0,0 +1,33 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Output.h"
+
+class Transient;
+
+class ProgressOutput : public Output
+{
+public:
+  static InputParameters validParams();
+
+  ProgressOutput(const InputParameters & parameters);
+
+protected:
+  void output() override;
+
+  Transient * _transient_executioner;
```

Comment:
```suggestion
  const Transient * const _transient_executioner;
```

Reply:
Yeah, tried that but most of the methods I'm calling on transient are not const and cannot be made const (like endTime which by design returns a writable reference). I could add a read only getter for each one of course...